devito/operations/interpolators.py

```diff
@@ -285,13 +288,18 @@ def callback():
                                                           field_offset=field_offset)
 
             # Substitute coordinate base symbols into the interpolation coefficients
-            eqns = [Inc(field.xreplace(vsub), _expr.xreplace(vsub) * b,
-                        implicit_dims=self.sfunction.dimensions)
-                    for b, vsub in zip(self._interpolation_coeffs, idx_subs)]
+            if increment:            
+                eqns = [Inc(field.xreplace(vsub), _expr.xreplace(vsub) * b,
+                            implicit_dims=self.sfunction.dimensions)
+                        for b, vsub in zip(self._interpolation_coeffs, idx_subs)]
+            else: 
+                eqns = [Eq(field.xreplace(idx_subs[0]), _expr.xreplace(idx_subs[0]),
```

Comment:
why are you picking `idx_subs[0]`? should you not traverse the whole list of `idx_subs` like in the `increment` case?

Could you do something like:

```
if increment:
   cls: Inc
else:
   cls: Eq
eqns = [cls(field.xreplace(....... .....) for b, vsub in zip(....)]
```
?

Reply:
because I thought in this case very specifically to apply boundary conditions, not source injection, for example. The ` ```increment=False``` option as I suggested should not be used for off-grid points.

Reply:
While this is true for your case, the generic case is still a standard potentially off the grid point injection without increment. So there is two option
- Use the injection as is knowing the coefficient will still be zero for the interpolation if the point is on the grid
- Define somehow an "on-the-grid" sparse function that ignores the interpolation.

The first case is definitely easier and should still work for your case and would boil down to what @FabioLuporini wrote earlier

Reply:
Indeed good point I missed that gotta think about it

Reply:
Getting back here since been looking into sparse, I think that the "best" way to go for `increment=Flase` is to do what `interpolation` does i.e 
- Define `sum` and do the standard `sum = interp(expr)`
- assign sum afterward `field = sum`

This would cover the off-the-grid case as well